src/btree/bt_sync.c

```diff
@@ -143,6 +142,29 @@ __sync_ref_mark_deleted(WT_SESSION_IMPL *session, WT_REF *ref)
     return (0);
 }
 
+/*
+ * __sync_ref_int_obsolete_cleanup --
+ *     Traverse the internal page and identify the leaf pages that are obsolete and mark them as
+ *     deleted.
+ */
+static int
+__sync_ref_int_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *intref)
+{
+    WT_PAGE_INDEX *pindex;
+    WT_REF *ref;
+    uint32_t slot;
+
+    WT_INTL_INDEX_GET(session, intref->page, pindex);
+    for (slot = 0; slot < pindex->entries; slot++) {
+        ref = pindex->index[slot];
+
+        if (__sync_ref_is_obsolete(session, ref))
+            WT_RET(__sync_ref_mark_deleted(session, ref));
```

Comment:
`__sync_ref_is_obsolete()` should ignore internal pages.

I'd move the rest of the tests there, too. It makes more sense that a function named "is obsolete" checks and returns false than a function named "mark deleted".

Reply:
Done.